src/unix/pipe.c

```diff
@@ -302,3 +302,59 @@ uv_handle_type uv_pipe_pending_type(uv_pipe_t* handle) {
   else
     return uv__handle_type(handle->accepted_fd);
 }
+
+
+int uv_pipe_chmod(uv_pipe_t* handle, int mode) {
+  unsigned desired_mode;
+  struct stat pipe_stat;
+  char* name_buffer;
+  size_t name_len;
+  int r;
+
+  if (!handle || uv__stream_fd(handle) == -1)
+    return UV_EBADF;
```

Comment:
Shouldn't the Unix code return `-EBADF`. Same for `UV_EINVAL` and `UV_ENOMEM` below.